components/widgets/resources/src/main/resources/resources/uicomponents/widgets/geneValidation.js

```diff
@@ -61,9 +66,14 @@ var XWiki = (function(XWiki) {
 
   var init = function(event) {
     ((event && event.memo.elements) || [$('body')]).each(function(element) {
-      element.select('input.gene-name').each(function(input) {
+      element.select('[name^="PhenoTips.GeneClass_"][name$="_gene"]').each(function(input) {
         if (!input.__Gene_validator) {
-          input.__Gene_validator = new XWiki.widgets.GeneValidator(input);
+          input.__Gene_validator = new XWiki.widgets.GeneVariantValidator(input);
+        }
+      });
+	  element.select('[name^="PhenoTips.GeneVariantClass_"][name$="_cdna"]').each(function(input) {
```

Comment:
Since the code to execute is almost identical in the two selectors, why don't you combine them? Like this: `element.select('selector 1', 'selector 2').each...`